basic-caching/README.md

```diff
@@ -0,0 +1,93 @@
+# Basic Caching
+
+This sample shows how to implement caching using Apigee's [ResponseCache](https://cloud.google.com/apigee/docs/api-platform/reference/policies/response-cache-policy), [PopulateCache](https://cloud.google.com/apigee/docs/api-platform/reference/policies/populate-cache-policy), and [LookupCache](https://cloud.google.com/apigee/docs/api-platform/reference/policies/lookup-cache-policy) policies.
+
+## About caching
+
+Using policies for general purpose caching, you can persist any objects your proxy requires across multiple request/response sessions. You can also cache the response of a backend resource with the ResponseCache policy. Response caching is especially helpful when backend data is updated only periodically. The ResponseCache policy can reduce calls to backend data sources.
+
+## How it works
+
+## Implementation on Apigee 
+
+This sample demonstrates how to use the ResponseCache policy to cache the response from the target backend. When the request cannot be served from a cache entry, the current time is also cached, and is returned as the cached-on header. Since this additional data point is not part of the target backend's response payload, caching is achieved by using the PopulateCache, and LookupCache policies.
+
+## Prerequisites
+1. [Provision Apigee X](https://cloud.google.com/apigee/docs/api-platform/get-started/provisioning-intro)
+2. Configure [external access](https://cloud.google.com/apigee/docs/api-platform/get-started/configure-routing#external-access) for API traffic to your Apigee X instance
+3. Access to deploy proxies in Apigee
+4. Make sure the following tools are available in your terminal's $PATH (Cloud Shell has these preconfigured)
+    * [gcloud SDK](https://cloud.google.com/sdk/docs/install)
+    * unzip
+    * curl
+    * jq
+    * npm
+# (QuickStart) Setup using CloudShell
+
+Use the following GCP CloudShell tutorial, and follow the instructions.
+
+[![Open in Cloud Shell](https://gstatic.com/cloudssh/images/open-btn.png)](https://ssh.cloud.google.com/cloudshell/open?cloudshell_git_repo=https://github.com/GoogleCloudPlatform/apigee-samples&cloudshell_git_branch=main&cloudshell_workspace=.&cloudshell_tutorial=basic-quota/docs/cloudshell-tutorial.md)
+
+## Setup instructions
+
+1. Clone the `apigee-samples` repo, and switch the `basic-caching` directory
+
+``` bash
+git clone https://github.com/GoogleCloudPlatform/apigee-samples.git
+cd apigee-samples/basic-caching
+```
+
+2. Set the following variables:
```

Comment:
Most of the other samples implement an `env.sh` file to house and variables, and then instruct the user to `source` the file. Might want to do that here just for consistency.

Reply:
Good point. Changed it in [d7e9eb6](https://github.com/GoogleCloudPlatform/apigee-samples/pull/67/commits/d7e9eb675143ccf44faba9ebd0505063f3ff8798).

Also noticed I didn't include text for the "How it works" section of the readme...fixed that as well.